### DISCUSSION TITLE
Events are deprecated, But docs encourage to write them

### DISCUSSION THREAD

**Discussion by Sarmad426** [fastapi/fastapi#12027] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

items = {}

# Startup event
@app.on_event("startup")
async def startup_event():
    items["foo"] = {"name": "Fighters"}
    items["bar"] = {"name": "Tenders"}


@app.get("/items/{item_id}")
async def read_items(item_id: str):
    return items[item_id]
```

### Description

When you connect to a database, we want to create database and tables when the application first starts. But the events are deprecated. When you visit the page for lifespan events at <https://fastapi.tiangolo.com/advanced/events/> the docs are encouraging to write those events. And you don't get any docs about if they are deprecated or not. Infact the docs are encouraging to write them. <https://fastapi.tiangolo.com/advanced/events/#startup-event>

### Operating System

Windows

### Operating System Details

It is not Operating system specific. Because it is just a warning. Although the events work absolutely fine. 

### FastAPI Version

0.110.0

### Pydantic Version

2.8.2

### Python Version

3.12

### Additional Context

![image](https://github.com/user-attachments/assets/74945f15-540d-4758-aff1-1f4d1b455bd7)

**Comment by sinisaos** [comment, marked as answer]
@Sarmad426 [Here](https://fastapi.tiangolo.com/advanced/events/#alternative-events-deprecated) is deprecated warning for events in docs.

**Comment by immanish0106** [comment]
It is deprecated, how can i still use it?

**Reply by Kludex** [reply]
Deprecation doesn't mean they were removed. Remotion is next step.

**Reply by nameer** [reply]
@immanish0106 If you are asking for the alternative, [lifespan](https://fastapi.tiangolo.com/advanced/events/#lifespan) is the answer.

**Comment by ramast** [comment]
Is it possible to  also update the SQL Databases docs? it also mentions on_event("startup")

https://fastapi.tiangolo.com/tutorial/sql-databases/

specifically this part

```
@app.on_event("startup")
def on_startup():
    create_db_and_tables()
```

Sorry I am a total beginner..
Also while at it could someone please recommend the right way to write this code ?

**Reply by sinisaos** [reply]
@ramast You should use [lifespan](https://fastapi.tiangolo.com/advanced/events/#lifespan). Here is an [example](https://github.com/sinisaos/headless-forum-fastapi/blob/master/main.py) of how Piccolo ORM opens and closes the connection pool in lifespan. Your example should be something like this.
```python
from contextlib import asynccontextmanager

from fastapi import FastAPI

@asynccontextmanager
async def lifespan(app: FastAPI):
    await create_db_and_tables()
    yield
    await drop_db_and_tables()

app = FastAPI(lifespan=lifespan)
```

**Reply by ramast** [reply]
Thanks a lot!
